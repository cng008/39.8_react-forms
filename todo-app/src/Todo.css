.Todo {
  display: flex;
  align-items: center;
  font-size: 1.2em;
}

.Todo-Input {
  font-size: 1.2em;
  margin-left: 1em;
  border: none;
}

button {
  border: 1px solid #dcdcdc;
  border-radius: 4px;
}

.fa-solid {
  color: #555555;
}

/* custom checkbox */
input[type='checkbox'] {
  position: relative;
  width: 2em;
  height: 2em;
  color: #363839;
  border: 1px solid #bdc1c6;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: 0;
  cursor: pointer;
  transition: background 175ms cubic-bezier(0.1, 0.1, 0.25, 1);
}
input[type='checkbox']::before {
  position: absolute;
  content: '';
  display: block;
  top: 2px;
  left: 7px;
  width: 8px;
  height: 14px;
  border-style: solid;
  border-color: #fff;
  border-width: 0 2px 2px 0;
  transform: rotate(45deg);
  opacity: 0;
}
input[type='checkbox']:checked {
  color: #fff;
  border-color: #06842c;
  background: #06842c;
}
input[type='checkbox']:checked::before {
  opacity: 1;
}
input[type='checkbox']:checked ~ label::before {
  -webkit-clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
  clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
}

label {
  position: relative;
  cursor: pointer;
  font-size: 1.5em;
  font-weight: 600;
  padding: 0 0.25em 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
label::before {
  position: absolute;
  content: attr(data-content);
  color: #9c9e9f;
  -webkit-clip-path: polygon(0 0, 0 0, 0% 100%, 0 100%);
  clip-path: polygon(0 0, 0 0, 0% 100%, 0 100%);
  text-decoration: line-through;
  text-decoration-thickness: 3px;
  -webkit-text-decoration-color: #363839;
  text-decoration-color: #363839;
  transition: -webkit-clip-path 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  transition: clip-path 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
  transition: clip-path 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94),
    -webkit-clip-path 200ms cubic-bezier(0.25, 0.46, 0.45, 0.94);
}
