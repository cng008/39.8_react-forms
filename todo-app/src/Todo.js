import React, { useState } from 'react'
import './Todo.css'

/** display a div with the task of the todo
 *
 *  * This has no state
 *
 */

const Todo = ({ id, todo, handleRemove }) => {
  const [isHovering, setIsHovering] = useState(false)
  const handleMouseOver = () => setIsHovering(true)
  const handleMouseOut = () => setIsHovering(false)
  const remove = () => handleRemove(id)

  return (
    <>
      <div
        className="Todo"
        data-testid="Todo"
        onMouseOver={handleMouseOver}
        onMouseOut={handleMouseOut}
      >
        <p>{todo}</p>
        <div
          className="productbox"
          onMouseEnter={handleMouseOver}
          onMouseLeave={handleMouseOut}
        >
          {isHovering && <button onClick={remove}>x</button>}
        </div>
      </div>
    </>
  )
}

export default Todo
