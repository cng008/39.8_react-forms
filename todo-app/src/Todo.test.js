import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Todo from './Todo'
import '@testing-library/jest-dom/extend-expect'

describe('<Todo /> rendering', () => {
  it('renders without crashing', function() {
    render(<Todo />)
  })

  it('matches snapshot', function() {
    const { asFragment } = render(<Todo />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot when editing', function() {
    const { asFragment, getByTestId } = render(<Todo />)
    const editButton = getByTestId('Edit')
    fireEvent.click(editButton)
    expect(asFragment()).toMatchSnapshot()
  })
})
