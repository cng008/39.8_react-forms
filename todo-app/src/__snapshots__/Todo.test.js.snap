// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    data-testid="Todo"
  >
    <input
      type="checkbox"
    />
    <p />
    <div
      class="Todo-Buttons"
      data-testid="Edit"
    />
    <div
      class="Todo-Buttons"
      data-testid="Remove"
    />
  </div>
</DocumentFragment>
`;

exports[`<Todo /> rendering matches snapshot when editing 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    data-testid="Todo"
  >
    <input
      type="checkbox"
    />
    <p />
    <div
      class="Todo-Buttons"
      data-testid="Edit"
    />
    <div
      class="Todo-Buttons"
      data-testid="Remove"
    />
  </div>
</DocumentFragment>
`;
